pkg/operator/controllers/machine/machine_controller.go

```diff
@@ -0,0 +1,77 @@
+package machine
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	machinev1beta1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/Azure/ARO-RP/pkg/operator"
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers"
+	_ "github.com/Azure/ARO-RP/pkg/util/scheme"
+)
+
+type MachineReconciler struct {
+	maocli                 maoclient.Interface
+	arocli                 aroclient.Interface
+	log                    *logrus.Entry
+	isLocalDevelopmentMode bool
+	role                   string
+}
+
+func NewMachineReconciler(log *logrus.Entry, maocli maoclient.Interface, arocli aroclient.Interface, isLocalDevelopmentMode bool, role string) *MachineReconciler {
+
+	return &MachineReconciler{
+		maocli:                 maocli,
+		arocli:                 arocli,
+		log:                    log,
+		isLocalDevelopmentMode: isLocalDevelopmentMode,
+		role:                   role,
+	}
+}
+
+func (r *MachineReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	// TODO(mj): Reconcile will eventually be receiving a ctx (https://github.com/kubernetes-sigs/controller-runtime/blob/7ef2da0bc161d823f084ad21ff5f9c9bd6b0cc39/pkg/reconcile/reconcile.go#L93)
+	ctx := context.TODO()
+
+	err := r.Check(ctx)
+	if err != nil {
+		r.log.Errorf("%d", err)
+	}
+
+	return reconcile.Result{RequeueAfter: time.Hour, Requeue: true}, err
```

Comment:
Still not sure if we want to do this once an hour or because we split this into its own controller for it to reconcile on its own by what it watches.  @mjudeikis 
Also, I believe you may want to get the cluster object before you do the `r.Check` when reconciling.  
`instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, request.Name, metav1.GetOptions{})`

Reply:
> 
> Also, I believe you may want to get the cluster object before you do the r.Check when reconciling.
> `instance, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, request.Name, metav1.GetOptions{})`
> 
>

Do we want the reconciler to do anything with the cluster object, or just ensure that it can be loaded in?

Reply:
is there something in the cluster object we need? If no, we should not be getting it here. 
Whats needs to change is that timer reconcile needs to be removed and we should reconcile each time Machine object changes (look how other controller enqueue on object changes)

Reply:
I've removed the time-based requeue in the reconciler, and tested to ensure that cluster object status is getting updated each time there's a change.